docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+- **description** - a description of what your template does.
+- **generator** - specify generator [specific configuration](https://www.asyncapi.com/docs/tools/generator/configuration-file).
+  - **renderer** - can either be `react` or `nunjucks`. In this case the generator will pass your template to the react render engine to generate the output.
+  - **apiVersion** - specifies which major version of the [Parser-API](https://github.com/asyncapi/parser-api) your template will use.
+  - **generator** - a string representing the generator version-range your template is compatible with.
+  - **supportedProtocols** - A list that specifies which protocols are supported by your template.
+- **dependencies** - specifies which version of [`@asyncapi/generator-react-sdk`](https://github.com/asyncapi/generator-react-sdk) should be used.
+
+Run the command `npm install` on your terminal to install the dependencies specified in `package.json`.
+
+### `index.js` file
+
+The index.js file is used to define the logic for your template. Inside the template folder, create an `index.js` file and add the code snippet below:
+
+```js
+//1
+import { File } from '@asyncapi/generator-react-sdk'
+//2
+export default function ({ asyncapi }) {
+  //3
+  return <File name="client.py">{asyncapi.info().title()}</File>
+}
+```
+
+The code snippet above does the following:
+
+1. Import the `generator-react-sdk` dependency.
+2. The `asyncapi` argument is an instance of the [AsyncAPI Parser](https://www.asyncapi.com/docs/tools/generator/parser). It will allow you to access the content of the asyncapi document in your template using helper functions.
+3. The `asyncapi.info().title()` is using the info() helper function to return the info object from the AsyncAPI document illustrated in the code snippet below:
+
+``` json
+info:
+  title: Temperature Service
+  version: 1.0.0
+  description: This service is in charge of processing all the events related to temperature.
+```
+
+`asyncapi.info().title()` will return Temperature Service.
+
+### Test using AsyncAPI CLI
+
+To see this in action, run `asyncapi generate fromTemplate test/fixtures/asyncapi.yml ./ -o test/project` command on your terminal. If successful, you'll see the message below on your terminal:
+
+``` cmd
+Generation in progress. Keep calm and wait a bit... done
+Check out your shiny new generated files at output.
+```
+
+And navigating to the **project** directory, you should see a **client.py** file; the only content is Temperature Service.
+
+Let's break down the previous command:
+
+- `asyncapi generate fromTemplate` is how you use AsyncAPI generator via the AsyncAPI CLI.
+- `asyncapi.yaml` points to your AsyncAPI document.
+- `./` specifies the location of your template.
+- `-o` determines where to output the result.
+
+## Creating a template
+
+We will create an MQTT-supported template that will generate a Python client from the template and the AsyncAPI document above.
```

Comment:
```suggestion
You will create an MQTT-supported template that will generate a Python client from the template and the AsyncAPI document above.
```

Reply:
best docs are when they are directed to reader. With blog posts it is different, but with docs it should be about the reader. Make sure you use it everywhere in the document